src/main/java/org/opensearch/security/dlic/rest/validation/RequestContentValidator.java

```diff
@@ -393,4 +543,227 @@ public String message() {
+        }
+        if (!allowedValues.contains(value)) {
+            throw new IllegalArgumentException("Invalid " + fieldName + " [" + value + "]. Allowed " + errorContext + ": " + allowedValues);
+        }
+    }
+
+    /**
+     * Validates a JSON object node with non-empty string values
+     */
+    public static void validateObjectWithStringValues(String fieldName, JsonNode node) {
+        if (node == null || node.isNull()) {
+            return;
+        }
+
+        if (!node.isObject()) {
+            throw new IllegalArgumentException(fieldName + " must be an object");
+        }
+
+        if (!node.fieldNames().hasNext()) {
+            throw new IllegalArgumentException(fieldName + " cannot be empty");
+        }
+
+        node.fields().forEachRemaining(entry -> {
+            JsonNode val = entry.getValue();
+            if (!val.isTextual() || val.asText().isEmpty()) {
+                throw new IllegalArgumentException(fieldName + " for key [" + entry.getKey() + "] must be a non-empty string");
+            }
+        });
+    }
+
+    /* ========================================================================
+     * Pre-built Field Validators for Common Use Cases
+     * ======================================================================== */
+
+    /**
+     * Creates a validator for principal values (users, roles, backend_roles, etc.)
+     * @param allowWildcard whether to allow "*" as a standalone value
+     */
+    public static FieldValidator principalValidator(boolean allowWildcard) {
+        return (fieldName, value) -> {
+            if (value instanceof String strValue) {
+                validateSafeValue(fieldName, strValue, MAX_STRING_LENGTH, allowWildcard);
+            }
+        };
+    }
+
+    /**
+     * Validator for path-like values (no whitespace allowed)
+     */
+    public static final FieldValidator PATH_VALIDATOR = (fieldName, value) -> {
+        if (value instanceof String strValue) {
+            validatePath(fieldName, strValue, MAX_STRING_LENGTH);
+        }
+    };
+
+    /**
+     * Validator for array entry counts (works with JsonNode arrays)
+     */
+    public static final FieldValidator ARRAY_SIZE_VALIDATOR = (fieldName, value) -> {
+        if (value instanceof JsonNode node) {
+            if (node.isArray() && node.size() > MAX_ARRAY_SIZE) {
+                throw new IllegalArgumentException("Array field [" + fieldName + "] exceeds maximum size of " + MAX_ARRAY_SIZE);
+            }
+        } else if (value instanceof Integer) {
+            int count = (Integer) value;
+            if (count > MAX_ARRAY_SIZE) {
+                throw new IllegalArgumentException("Array field [" + fieldName + "] exceeds maximum size of " + MAX_ARRAY_SIZE);
+            }
+        }
+    };
+
+    /**
+     * Creates a validator that checks if a string value is in an allowed set
+     */
+    public static FieldValidator allowedValuesValidator(Set<String> allowedValues, String errorMessage) {
```

Comment:
Can we overload this method to avoid having to pass `null` in as the second param?

Reply:
sure